src/components/Modals/Nfts/NftModal.tsx

```diff
@@ -0,0 +1,282 @@
+import type { TabProps } from '@chakra-ui/react'
+import {
+  Box,
+  Button,
+  Flex,
+  Image,
+  Link,
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+  Skeleton,
+  Stat,
+  StatGroup,
+  StatLabel,
+  StatNumber,
+  Tab,
+  TabIndicator,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Tag,
+  TagLeftIcon,
+  useColorModeValue,
+  useMediaQuery,
+} from '@chakra-ui/react'
+import { useMemo, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { ArrowRightUp } from 'components/Icons/ArrowRightUp'
+import { DiamondIcon } from 'components/Icons/DiamondIcon'
+import { RawText } from 'components/Text'
+import { ordinalSuffix } from 'context/WalletProvider/NativeWallet/components/NativeTestPhrase'
+import { useModal } from 'hooks/useModal/useModal'
+import type { V2ZapperNft } from 'state/apis/zapper/client'
+import { getMediaType } from 'state/apis/zapper/client'
+import { useGetZapperCollectionsQuery } from 'state/apis/zapper/zapperApi'
+import { selectWalletAccountIds } from 'state/slices/common-selectors'
+import { useAppSelector } from 'state/store'
+import { breakpoints } from 'theme/theme'
+
+import { NftCollection } from './components/NftCollection'
+import { NftOverview } from './components/NftOverview'
+
+const NftTab: React.FC<TabProps> = props => {
+  const activeTabColor = useColorModeValue('blue.500', 'white')
+  return (
+    <Tab
+      color='gray.500'
+      fontWeight='medium'
+      px={0}
+      _selected={{ color: activeTabColor }}
+      {...props}
+    />
+  )
+}
+
+export type NftModalProps = {
+  zapperNft: V2ZapperNft
+}
+
+export const NftModal: React.FC<NftModalProps> = ({ zapperNft }) => {
+  const { nft } = useModal()
+  const { close, isOpen } = nft
+  const translate = useTranslate()
+  const [isMediaLoaded, setIsMediaLoaded] = useState(false)
+  const modalBg = useColorModeValue('white', 'gray.800')
+  const modalHeaderBg = useColorModeValue('gray.50', 'gray.785')
+  const [isLargerThanMd] = useMediaQuery(`(min-width: ${breakpoints['md']})`, { ssr: false })
+  const accountIds = useAppSelector(selectWalletAccountIds)
+
+  const collectionAddresses = useMemo(
+    () => (zapperNft?.collection?.address ? [zapperNft?.collection?.address] : []),
+    [zapperNft?.collection?.address],
+  )
+  const { data: zapperCollection } = useGetZapperCollectionsQuery(
+    { accountIds, collectionAddresses },
+    { skip: !collectionAddresses?.length },
+  )
+
+  const mediaUrl = zapperNft?.medias?.[0]?.originalUrl
```

Comment:
Is there ever a case where NFTs can have multiple medias i.e we will need a carousel in the future? If so we should add a TODO